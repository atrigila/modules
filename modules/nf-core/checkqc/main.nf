process CHECKQC {
    tag "$meta.id"
    label 'process_single'

    container "community.wave.seqera.io/library/python_numpy_pip_checkqc_interop:b5301d9801b8e66b"

    input:
    tuple val(meta), path(run_dir)
    path(checkqc_config)

    output:
    tuple val(meta), path("*checkqc_report.json"), emit: report
    tuple val(meta), path("*checkqc_log.txt")    , emit: log
    path "versions.yml"                          , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    // Exit if running this module with -profile conda / -profile mamba
    if (workflow.profile.tokenize(',').intersect(['conda', 'mamba']).size() >= 1) {
        error "CheckQC module does not support Conda yet. Please use Docker / Singularity / Podman instead."
    }

    def args = task.ext.args ?: ''
    def config = checkqc_config ? "--config $checkqc_config" : ''

    """
    checkqc \
        $args \
        $config \
        --json \
        $run_dir > checkqc_report.json || true

    cp .command.log checkqc_log.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        checkqc: \$( checkqc --version | sed -e "s/checkqc, version //g" )
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    """
    touch checkqc_report.json

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        checkqc: \$( checkqc --version | sed -e "s/checkqc, version //g" )
    END_VERSIONS
    """
}
