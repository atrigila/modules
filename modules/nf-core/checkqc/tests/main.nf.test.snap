{
    "homo_sapiens illumina test_interop_bcl2fastqstats_flowcell - QC Error": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_no_downgrade_errors"
                        },
                        "checkqc_report.json:md5,c0c5655b8f09d848de97f79073b2eae0"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test_no_downgrade_errors"
                        },
                        "checkqc_log.txt:md5,687d57a112fda6f56755d45e5c9cd0be"
                    ]
                ],
                "2": [
                    "versions.yml:md5,7c921c68a4de1cc3345edb1fd88444d1"
                ],
                "log": [
                    [
                        {
                            "id": "test_no_downgrade_errors"
                        },
                        "checkqc_log.txt:md5,687d57a112fda6f56755d45e5c9cd0be"
                    ]
                ],
                "report": [
                    [
                        {
                            "id": "test_no_downgrade_errors"
                        },
                        "checkqc_report.json:md5,c0c5655b8f09d848de97f79073b2eae0"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,7c921c68a4de1cc3345edb1fd88444d1"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-07-26T16:12:27.425381414"
    },
    "homo_sapiens illumina test_miseq_noRunParameters - Log Error: no File": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_no_xml"
                        },
                        "checkqc_report.json:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test_no_xml"
                        },
                        "checkqc_log.txt:md5,b63cf133c0110302ec57dc1ae25e2ef7"
                    ]
                ],
                "2": [
                    "versions.yml:md5,7c921c68a4de1cc3345edb1fd88444d1"
                ],
                "log": [
                    [
                        {
                            "id": "test_no_xml"
                        },
                        "checkqc_log.txt:md5,b63cf133c0110302ec57dc1ae25e2ef7"
                    ]
                ],
                "report": [
                    [
                        {
                            "id": "test_no_xml"
                        },
                        "checkqc_report.json:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,7c921c68a4de1cc3345edb1fd88444d1"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-07-26T16:11:59.031624784"
    },
    "homo_sapiens illumina test_interop_bcl2fastqstats_flowcell - QC OK": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "checkqc_report.json:md5,7e3ac9ce7163436cbd5f5420ca08ffcd"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "checkqc_log.txt:md5,3e196f53b93d614580436bc574daea45"
                    ]
                ],
                "2": [
                    "versions.yml:md5,7c921c68a4de1cc3345edb1fd88444d1"
                ],
                "log": [
                    [
                        {
                            "id": "test"
                        },
                        "checkqc_log.txt:md5,3e196f53b93d614580436bc574daea45"
                    ]
                ],
                "report": [
                    [
                        {
                            "id": "test"
                        },
                        "checkqc_report.json:md5,7e3ac9ce7163436cbd5f5420ca08ffcd"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,7c921c68a4de1cc3345edb1fd88444d1"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-07-26T16:11:32.798413187"
    }
}